Document external PostgreSQL earlier in the installation UX (#3783)
theforeman/foreman-documentation: guides/common/modules/proc_preparing-for-using-external-databases.adoc

Author: @ekohl , from github.com/theforeman/foreman-documentation/pull/2396 , I understand that the migration to the external DB is/was not possible on Debian, so I ifdef'd it out ( https://github.com/theforeman/foreman-documentation/pull/3783/commits/95eebc0ed0205a664cee8e5faa3247acfa0387d5 ). Was there something else that should be amended in the content, as far as you can tell?

Reviewer: There's no technical reason it's impossible. It's just that we never wrote up the correct instructions. Rather than presenting wrong instructions, it was ifdef'd out.

[id="Preparing_for_using_external_databases_{context}"]
= Preparing {Project} for using external databases

By default, the {Project} installation process includes installing a PostgreSQL database on the same host as {ProjectServer}.

> However, in certain {Project} deployments, using external databases instead of the default local databases can help with the server load.

Reviewer: This is probably not nicely worded, but there are also organizations with dedicated database admins.
```suggestion
However, in certain {Project} deployments, using external databases instead of the default local databases is desirable.
For example, to help with the server load or in organizations with a specialized database team.
```

.Prerequisite
* You have considered whether using an external database is beneficial for your use case. 
For more information, see {AdministeringDocURL}postgresql-as-an-external-database-considerations_admin[PostgreSQL as an external database considerations] in _{AdministeringDocTitle}_.

.Procedure
. Set up a system to host the external database. 
For instructions, see {AdministeringDocURL}preparing-a-host-for-external-databases_admin[Preparing a host for external databases] in _{AdministeringDocTitle}_.

. Install PostgreSQL on the external database host you prepared. 
For instructions, see {AdministeringDocURL}installing-postgresql_admin[Installing PostgreSQL] in _{AdministeringDocTitle}_.

.Next steps
* To set up an external database when installing {Project}, see xref:configuring-server_{project-context}[].

> * If you want to migrate your existing {Project} deployment to an external database, see {AdministeringDocURL}Migrating_from_Internal_Databases_to_External_Databases_admin[Migrating from internal {Project} databases to external databases] in _{AdministeringDocTitle}_.

Reviewer: This isn't really a next step, but perhaps an additional resource?

[id="Preparing_for_using_external_databases_{context}"]
= Optional: Preparing {Project} for using external databases

By default, the {Project} installation process includes installing a PostgreSQL database on the same host as {ProjectServer}.
However, in certain {Project} deployments, using external databases instead of the default local databases can help with the server load.

You can set up your {Project} deployment to use an external PostgreSQL database during the installation of {Project}. 
You can also migrate to an external PostgreSQL database on an existing {Project} deployment.

> 

Reviewer 2: What would you think about dropping these two lines? Line 7 doesn't add anything new -- that information is already expressed through the section heading, the intro, and the steps themselves. And line 8 is not relevant here because at this point in the process, users are preparing for the installation; these users don't need to know what to do when they already have an existing installation. Moreover, line 8 duplicates line 27.

Dropping these lines would help keep the installation prep content shorter, which is one of the goals of the content journey.

.Procedure

> 

Reviewer 2: ```suggestion
```
This applies to the other discreet headings as well. A blank line is unnecessary.

. Consider whether using an external database is beneficial for your use case. 

> For more information, see {AdministeringDocURL}postgresql-as-an-external-database-considerations_admin[PostgreSQL as an external database considerations] in _{AdministeringDocTitle}_.

Reviewer 2: How about turning this into a prerequisite?

Author: Well, it would make the procedure somewhat shorter (though it's already pretty short, I think), but it might be a bit awkward information flow-wise. I'll give it a shot in the next iteration, nonetheless.

> . If you want to use an external external PostgreSQL database for your {Project} deployment, prepare your {Project} environment as follows:

Reviewer 2: ```suggestion
. If you want to use an external PostgreSQL database for your {Project} deployment, prepare your {Project} environment as follows:
```

Reviewer 2: But actually, if you turn the previous step into a prerequisite, you should be able to drop this line and then only keep lines 17-21 as the actual steps. It would make the procedure shorter and simpler, which is a big plus IMO.

.. Set up a system to host the external database. 
For instructions, see {AdministeringDocURL}preparing-a-host-for-external-databases_admin[Preparing a host for external databases] in _{AdministeringDocTitle}_.

.. Install PostgreSQL on the external database host you prepared. 
For instructions, see {AdministeringDocURL}installing-postgresql_admin[Installing PostgreSQL] in _{AdministeringDocTitle}_.

.Next steps

> * To set up an external databases when installing {Project}, see xref:configuring-server_{project-context}[].

Reviewer 2: I'm not entirely sure whether this line is needed. I think all users will eventually reach the xref'ed section, won't they? Do we need to link to it if they will?

Author: Well, what I wanted to distinguish is the options to migrate to an external database while installing Satellite, as opposed to on an already installed environment. 

Also, assuming that the reader reads any other specific part of content before/after is generally discouraged, and even if the xref might be extraneous for some readers, for others it would still be helpful that they can click the link and see the instructions.

So all in all, I'd keep it as is :-)

> 

Reviewer 2: ```suggestion
```
